torch/_inductor/lowering.py

```diff
@@ -6099,6 +6098,27 @@ def fill_(x, fill_value):
     return mutate_to(x, full_like(x, fill_value))
 
 
+@register_lowering(prims.fill, type_promotion_kind=None)
+def prims_fill(x, fill_value):
+    dtype = x.get_dtype()
+    return Pointwise.create(
+        device=x.get_device(),
+        dtype=dtype,
+        inner_fn=lambda _: ops.constant(fill_value, dtype),
+        ranges=list(x.get_size()),
+    )
+
+
+@register_lowering(prims.full, type_promotion_kind=None)
+def prims_full(shape, fill_value, *, dtype, device, requires_grad):
+    return Pointwise.create(
```

Comment:
Could we do the full strides preservation and a potential prim op as separate prs ? Without a good reason, I would prefer prevent churn here. 

See, pattern for `full` here https://github.com/pytorch/pytorch/blob/e5afbe31245287a92fe328c404b3557e5c5eca73/torch/_inductor/fx_passes/post_grad.py#L702-L707

and here https://github.com/pytorch/pytorch/blob/e5afbe31245287a92fe328c404b3557e5c5eca73/torch/_inductor/fx_passes/serialized_patterns/_sfdp_pattern_17.py#L39